### ISSUE TITLE
Future: transition from bower to npm / yarn

### ISSUE THREAD

**Issue by gorhgorh** [adaptlearning/adapt_framework#2526]
### transition from bower to npm / yarn
notes following a discussion on the gitter channel, where we discussed advantages / need to choose between npm and yarn.
here are some questions that @oliverfoster raised 

- is yarn going to be the defacto standard npm client at some point?
- are the features going to work their way back into npm?
- does yarn have it's own module listing infrastructure or does it rely on the npm catalogue?
- would i need to go to two places to understand adapt's plugin management architecture (the npm and yarn websites) and is that tooling ambiguity worth the extra features?
- if we use npm does that stop us using yarn in the future and vice-versa?

here is my own take on these, feel free to discuss
#### tldr
I don't think adapt should get bound to npm or yarn only, currently yarn is 100% compatible with npm so stick to what npm offer, as it allow 100% of the npm OR yarn only users to use the framework.
One maybe interesting thing in yarn would be [workspaces](https://yarnpkg.com/en/docs/workspaces)

As for namespacing, the adapt org is still up as I'm writing this issue.
so you can have `@adapt/mcq` like package names.

more detailed answers bellow

#### is yarn going to be the defacto standard npm client at some point?
i don't think so, but big things will happen soon in the registry world as the revolution against npm began (almost all of the historial employee are now out, some of them rolling their own infra)
#### are the features going to work their way back into npm?
no and none of them really wants that, as they want to captivate their users ;(
#### does yarn have it's own module listing infrastructure or does it rely on the npm catalogue?
it rely on npm infra, as most of the current replacement for npm, some alternative package registry will come soon, but will be drop in replacement of npm (entropic, github have something in mind, too)
#### would i need to go to two places to understand adapt's plugin management architecture (the npm and yarn websites) and is that tooling ambiguity worth the extra features?
for now, I don't think so, [workspaces](https://yarnpkg.com/en/docs/workspaces) seems like something adapt can use, but the price is too high imo

#### if we use npm does that stop us using yarn in the future and vice-versa?
As of today, only yarn have yarn specific feature that locks in.

**Comment by oliverfoster** [comment]
duplicate
